src/app/courses/courses.component.ts

```diff
@@ -56,45 +57,33 @@ export class CoursesComponent implements OnInit, AfterViewInit {
   userId = this.userService.get()._id;
   userShelf: any = [];
   private onDestroy$ = new Subject<void>();
+  getOpts = this.parent ? { domain: this.userService.getConfig().parentDomain } : {};
 
   constructor(
     private couchService: CouchService,
     private dialog: MatDialog,
     private planetMessageService: PlanetMessageService,
     private router: Router,
     private route: ActivatedRoute,
-    private userService: UserService
+    private userService: UserService,
+    public coursesService: CoursesService
```

Comment:
Do not make a service public. Expose the methods you need within component methods.